src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/create-exercise-unit/create-exercise-unit.component.ts

```diff
@@ -18,25 +18,18 @@ import { faSort, faTimes } from '@fortawesome/free-solid-svg-icons';
     styleUrls: ['./create-exercise-unit.component.scss'],
 })
 export class CreateExerciseUnitComponent implements OnInit {
-    @Input()
-    hasCancelButton: boolean;
-    @Input()
-    hasCreateExerciseButton: boolean;
-    @Input()
-    shouldNavigateOnSubmit = true;
-    @Input()
-    lectureId: number | undefined;
-    @Input()
-    courseId: number | undefined;
-    @Input()
-    currentWizardStep: number;
+    protected readonly faTimes = faTimes;
+    protected readonly faSort = faSort;
 
-    @Output()
-    onCancel: EventEmitter<any> = new EventEmitter<any>();
-    @Output()
-    onExerciseUnitCreated: EventEmitter<any> = new EventEmitter<any>();
+    @Input() hasCancelButton: boolean;
```

Comment:
is there a reason why we do not use input.required here if we want to migrate to signals?

Reply:
The reason is that I need to refactor quite some components throughout the lecture view to accomplish adding a status bar to the lecture edit mode https://github.com/ls1intum/Artemis/pull/9655, as I want to prevent these refactorings from bloating up the scope. I only touched the necessary parts for the status bar PR.

In this example e.g. verifying the usages of all inputs and outputs, retesting the functionality, and adjusting the PR description cost me additional 40 minutes 😅